nemo/collections/nlp/modules/common/bert_module.py

```diff
@@ -37,8 +37,21 @@ def input_types(self) -> Optional[Dict[str, NeuralType]]:
     def output_types(self) -> Optional[Dict[str, NeuralType]]:
         return {"last_hidden_states": NeuralType(('B', 'T', 'D'), ChannelType())}
 
+    @classmethod
+    def restore_from(cls, restore_path: str):
+        """Restores module/model with weights"""
+        pass
+
+    @classmethod
     def save_to(self, save_path: str):
+        """Saves module/model with weights"""
         pass
 
-    def restore_from(self, restore_path: str):
-        self.load_state_dict(torch.load(restore_path))
+    def restore_weights(self, restore_path: str):
+        """Restores module/model's weights"""
```

Comment:
Let's leave the existing implementation here: self.load_state_dict(torch.load(restore_path)) - so that it only works for hf models and you're overriding this in the megatron

Reply:
> Just minor comments + Jenkins test for at least pretrained_model_name megatron-bert-345m-uncased is still missing.
> Is there a timeframe for the APEX related changes in the Megatron-LM? This PR will break code for NeMo users without APEX, I'd recommend adding apex checks for now.

APEX changes were just merged in megatron-LM.

Reply:
pushed megatron-lm version 1.1 to pypi.org with the recent changes